chat.sympl

```diff
@@ -183,177 +228,260 @@ def _guard_input(input_description: str, input_: str) -> None:
     if len(input_) > 4000:
         cvm.error(f"{input_description} cannot be longer than 4000 characters.")
 
-@public_clientside
+@clientside_helper
 def _create_room(room_name: str) -> None:
+    #this function will run checks on the propsed
+    #room name and cvm.error if there are issues
+    #cvm.error will end the transaction
     _guard_input("Room name", room_name)
     room_channel = cvm.new_channel('RID')
     with PostTxArgs(room_channel):
         _create_room_execute(room_name, room_channel)
-    cvm.job_start()
-
 
 @executable
-def _create_room_execute(room_name: str, room_channel: ChannelName) -> None:
+def _create_room_execute(room_name: str, room_channel: ChannelName) -> CreateRoomEvent:
```

Comment:
I did this because of the "migrating job complete" docs https://iportal.symbiont.io/docs/assembly/develop/sailfish/references/lang7_migration_4.2.0.html?highlight=lang7%20migration#migrating-cvm-job-completed

![image](https://user-images.githubusercontent.com/87322808/128386566-5bf633db-9189-453e-ab2a-1a0b9853e4b3.png)

And what the notion page said:
https://www.notion.so/symbiontio/Chat-contract-showcase-improvements-50527d40e8e14f8ca03a565e698b59a4#04ce98e1f8544688bb854ae7c84a6fa1

Reply:
well, it's hard to argue with this! :) Let's not argue about this in this PR, then. But this is a discussion we need to have with the language team in general @symbiont-larry-diehl

Reply:
this is inherited functionality/behavior ported to Haskell VM from existing Python VM courtesy of @symbiont-eric-torreborre when the work was done to remove the need for explicit `job_complete`'s. it could be revisited if people don't like it but so far i haven't thought much about it as it was only inherited from Lang 7